tool/tctl/common/plugin/awsic_test.go

```diff
@@ -315,3 +320,202 @@ func TestUseSystemCredentialsInput(t *testing.T) {
+			expectUpdate:        true,
+			updateError:         trace.NotFound("no such credential"),
+			assertError:         require.Error,
+		},
+		{
+			name: "validation failure",
+			cliArgs: awsICRotateCredsArgs{
+				pluginName:    types.PluginTypeAWSIdentityCenter,
+				validateToken: true,
+				payload:       "some-token",
+			},
+			expectValidation:    true,
+			validationError:     trace.AccessDenied("Validation failed"),
+			pluginValueProvider: validAWSICPlugin,
+			expectUpdate:        false,
+			assertError:         require.Error,
+		},
+		{
+			name: "bypass validation",
+			cliArgs: awsICRotateCredsArgs{
+				pluginName:    types.PluginTypeAWSIdentityCenter,
+				validateToken: false,
+				payload:       "some-token",
+			},
+			expectValidation:    false,
+			validationError:     trace.AccessDenied("Validation failed"),
+			pluginValueProvider: validAWSICPlugin,
+			expectUpdate:        true,
+			assertError:         require.NoError,
+		},
+	}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			cliArgs := PluginsCommand{
+				rotateCreds: pluginRotateCredsArgs{
+					awsic: test.cliArgs,
+				},
+			}
+
+			pluginsClient := &mockPluginsClient{}
+			pluginsClient.
+				On("GetPlugin", anyContext, mock.Anything, mock.Anything).
+				Run(func(args mock.Arguments) {
+					req, ok := args.Get(1).(*pluginsv1.GetPluginRequest)
+					require.True(t, ok, "expecting a *pluginsv1.GetPluginRequest, got %T", args.Get(1))
+					require.Equal(t, test.cliArgs.pluginName, req.Name)
+					require.True(t, req.WithSecrets)
+				}).
+				Return(test.pluginValueProvider(), test.pluginFetchError)
+
+			if test.expectUpdate {
+				pluginsClient.
+					On("UpdatePluginStaticCredentials", anyContext, mock.Anything, mock.Anything).
+					Return(func(ctx context.Context, in *pluginsv1.UpdatePluginStaticCredentialsRequest, _ ...grpc.CallOption) (*pluginsv1.UpdatePluginStaticCredentialsResponse, error) {
+						q := in.GetQuery()
+						require.NotNil(t, q, "Update request must specify target labels")
+						require.NotEmpty(t, q.Labels, "Update request must specify non-empty labels")
+
+						return &pluginsv1.UpdatePluginStaticCredentialsResponse{
+							Credential: &types.PluginStaticCredentialsV1{Spec: in.GetCredential()},
+						}, test.updateError
+					})
+			}
+
+			scimClient := mockSCIMClient{
+				Client:       scimsdk.NewSCIMClientMock(),
+				pingResponse: test.validationError,
+			}
+
+			args := installPluginArgs{
+				plugins: pluginsClient,
+				scimProvider: func(_, _, token string) (scimsdk.Client, error) {
+					require.Equal(t, test.cliArgs.payload, token)
+					return &scimClient, nil
+				},
+			}
+
+			err := cliArgs.RotateAWSICCreds(context.Background(), args)
+			test.assertError(t, err)
+
+			pluginsClient.AssertExpectations(t)
+			require.Equal(t, test.expectValidation, scimClient.pingCalled,
+				"SCIM validation Ping expected: %t, Ping called %t", test.expectValidation, scimClient.pingCalled)
```

Comment:
I am confused, you mentioned that validation is now reverted but we have test here ? 
https://github.com/gravitational/teleport/pull/57691#discussion_r2321133429